regen/embed.pl

```diff
@@ -59,7 +69,5440 @@ BEGIN
 $has_compat_macro{$_} = 1 for @have_compatibility_macros;
 my %perl_compats;   # Have 'perl_' prefix
 
+# This program inspects various top-level header files, except those on this
+# list.
+my %skip_files;
+$skip_files{$_} = 1 for qw(
+                            charclass_invlists.inc
+                            config.h
+                            embed.h
+                            fakesdio.h
+                            nostdio.h
+                            perl_langinfo.h
+                            perlio.h
+                            proto.h
+                            XSUB.h
+                          );
+
+# Items that are marked as being in the API are, by definition, not namespace
+# pollutants.  To find those, this program looks for API declarations.  These
+# are in embed.fnc, any top-level dot c file, and certain pod files.  In order
+# to save a bit of run time, this static list comprises the pod files known to
+# have such defintions.  It is not expected this will change often.
+my @pod_list = qw(
+                   INSTALL
+                   pod/perlguts.pod
+                   pod/perlinterp.pod
+                   pod/perlapio.pod
+                   pod/perlhacktips.pod
+                   pod/perlcall.pod
+                   pod/perlreguts.pod
+                   pod/perlmroapi.pod
+                   pod/perlembed.pod
+                   dist/ExtUtils-ParseXS/lib/perlxs.pod
+                   pod/perliol.pod
+                       pod/perlreapi.pod
+                 );
+
+# This is a list of symbols that are not documented to be available for
+# modules to use, but are nevertheless currently not kept by embed.h from
+# being visible to the world.
+#
+# Strive to make this list empty.
+#
+# The list does not include symbols that we have documented as being reserved
+# for perl's use, namely those that begin with 'PL_' or contain qr/perl/i.
+# There are two parts of the list; the second part contains the symbols which
+# have a trailing underscore; the first part those without.
 #
+# For all modules that aren't deliberating using particular names, all the
+# other symbols on it are namespace pollutants.
+
+my @unexpectedly_used_outside_core = qw(
```

Comment:
`@unexpectedly_used_outside_core` seems like a bad name, many symbols are API and expected to be used outside of core (like `AMGf_assign`) and many don't seem to be used outside of core at all.